tests/functional/modules/test_vault_pki.py

```diff
@@ -0,0 +1,508 @@
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
+
+import pytest
+from cryptography import x509
+from cryptography.hazmat.primitives import serialization
+from salt.utils.x509 import generate_rsa_privkey
+from salt.utils.x509 import load_cert
+
+from saltext.vault.utils.vault.exceptions import VaultException
+from saltext.vault.utils.vault.pki import dec2hex
+from tests.support.vault import vault_delete
+from tests.support.vault import vault_disable_secret_engine
+from tests.support.vault import vault_enable_secret_engine
+from tests.support.vault import vault_list
+from tests.support.vault import vault_list_detailed
+from tests.support.vault import vault_read
+from tests.support.vault import vault_write
+
+pytest.importorskip("docker")
+
+pytestmark = [
+    pytest.mark.slow_test,
+    pytest.mark.skip_if_binaries_missing("vault", "getent"),
+    pytest.mark.usefixtures("vault_container_version"),
+]
+
+
+@pytest.fixture(scope="module")
+def minion_config_overrides(vault_port):
+    return {
+        "vault": {
+            "auth": {
+                "method": "token",
+                "token": "testsecret",
+            },
+            "cache": {
+                "backend": "disk",  # ensure a persistent cache is available for get_creds
+            },
+            "server": {
+                "url": f"http://127.0.0.1:{vault_port}",
+            },
+        }
+    }
+
+
+@pytest.fixture
+def testrole():
+    return {"ttl": 3600, "max_ttl": 86400, "allow_any_name": True, "issuer_ref": "testissuer"}
+
+
+@pytest.fixture
+def testissuer():
+    return {"issuer_name": "testissuer", "common_name": "Test Issuer CA"}
+
+
+@pytest.fixture
+def testissuer2():
+    return {"issuer_name": "testissuer2", "common_name": "Test Issuer CA 2"}
+
+
+@pytest.fixture(scope="module")
+def private_key():
+    pk = generate_rsa_privkey(2048)
+    pk_encoding = getattr(serialization.Encoding, "PEM")
+    pk_bytes = pk.private_bytes(
+        pk_encoding,
```

Comment:
nitpick: Sorry, this was not clear from the previous larger comment.

```suggestion
    pk_bytes = pk.private_bytes(
        serialization.Encoding.PEM,
```